holidays/financial/_sifma.py

```diff
@@ -0,0 +1,164 @@
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from holidays.calendars.gregorian import DEC, FRI, JUL, MAY, MON, SAT, SUN, THU, TUE, WED, date
+from holidays.constants import HALF_DAY, PUBLIC
+from holidays.groups import ChristianHolidays, InternationalHolidays
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    ObservedRule,
+    SAT_TO_PREV_FRI,
+    SUN_TO_NEXT_MON,
+)
+
+# Custom observed rule for early close dates.
+# Applied directly to the actual holiday date to determine if there's an early close.
+# MON: holiday on Monday → early close on Friday (-3)
+# TUE-FRI: holiday on Tue-Fri → early close the day before (-1)
+# SAT: holiday on Saturday → early close on Friday (-2)
+# SUN: holiday on Sunday → early close on Friday (-2)
+SIFMA_EARLY_CLOSE = ObservedRule({MON: -3, TUE: -1, WED: -1, THU: -1, FRI: -1, SAT: -2, SUN: -2})
+
+
+class SIFMAHolidays(ObservedHolidayBase, ChristianHolidays, InternationalHolidays):
+    """Base class for SIFMA (Securities Industry and Financial Markets Association) holidays.
+
+    SIFMA provides recommendations for full and early market closures for the trading of
+    U.S. dollar-denominated government securities, mortgage- and asset-backed securities,
+    over-the-counter investment-grade and high-yield corporate bonds, municipal bonds,
+    and secondary money market trading.
+
+    This base class implements the common holidays observed across SIFMA-recommended
+    calendars, including both full closures (PUBLIC category) and early closes
+    (HALF_DAY category).
+
+    References:
+        * <https://www.sifma.org/resources/general/holiday-schedule/>
+        * <https://web.archive.org/web/20250210040000/https://www.sifma.org/resources/general/holiday-schedule/>
+
+    Historical references:
+        * <https://web.archive.org/web/20250210040000/https://www.sifma.org/resources/general/us-holiday-archive/>
+    """
+
+    observed_label = "%s (observed)"
+    supported_categories = (HALF_DAY, PUBLIC)
+    start_year = 1950
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        kwargs.setdefault("observed_rule", SAT_TO_PREV_FRI + SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._move_holiday(self._add_new_years_day("New Year's Day"))
+
+        # Martin Luther King Jr. Day (3rd Monday of January).
+        # Established as federal holiday in 1986, bond markets observed since 1998.
```

Comment:
```suggestion
        # Martin Luther King Jr. Day.
```
And the same for all holidays below (no need for such comments, except Good Friday note).